src/Text/Layout/Table/Primitives/ColumnModifier.hs

```diff
@@ -96,15 +97,25 @@ columnModifier pos cms colModInfo = case colModInfo of
 -- | Derive the 'ColModInfo' for each column of a list of rows by using the
 -- corresponding specifications.  See 'deriveColModInfoFromColumn' for details.
 deriveColModInfos :: Cell a => [(LenSpec, AlignSpec)] -> [Row a] -> [ColModInfo]
-deriveColModInfos specs = zipWith ($) (fmap deriveColModInfoFromColumn specs) . transpose
+deriveColModInfos specs = deriveColModInfosFromColumns specs . transpose
+
+deriveColModInfos' :: Cell a => [ColSpec] -> [Row a] -> [ColModInfo]
```

Comment:
I know that these are internal, but the naming still confuses me. `deriveColModInfos`, `deriveColModInfos'`, `deriveColModInfoFromColumns`, `deriveColModInfoFromColumn` are very similar, and it's a bit tricky to know when to use which. The types help, but maybe add a few lines of documentation to help distinguish them?

Reply:
Good point.  I renamed them completely to avoid any confusion.